tensorflow/python/ops/array_ops.py

```diff
@@ -125,6 +125,32 @@ def shape(input, name=None):
     else:
       return gen_array_ops.shape(input, name=name)
 
+def size(input, name=None):
+  """Returns the size of a tensor.
+
+  This operation returns an integer representing the number of elements in
+  `input`.
+
+  For example:
+
+  ```python
+  # 't' is [[[1, 1,, 1], [2, 2, 2]], [[3, 3, 3], [4, 4, 4]]]]
+  size(t) ==> 12
+  ```
+
+  Args:
+    input: A `Tensor` or `SparseTensor`.
+    name: A name for the operation (optional).
+
+  Returns:
+    A `Tensor` of type `int32`.
+  """
+  with ops.op_scope([input], name, "Size") as name:
+    if isinstance(input, ops.SparseTensor):
+      return gen_math_ops._prod(input.shape, 0, name=name)
```

Comment:
Agreed with @yaroslavvb.

One minor thing: `input.shape` for `SparseTensor` is actually int64, so to avoid inconsistency we should either (1) change the "Returns:" part in our documentation, or (2) cast to int32 here.  

I am leaning towards (2) to keep it simple.  @ebrevdo any thoughts?